lib/rust/flexer/src/automata/state.rs

```diff
@@ -106,9 +72,56 @@ impl From<Vec<(RangeInclusive<u32>, usize)>> for State {
         let link = |(range, id): (RangeInclusive<u32>, usize)| {
             let start = Symbol{val:*range.start()};
             let end   = Symbol{val:*range.end()};
-            Link {symbols: start..=end, target: Id{ id }}
+            Transition {symbols: start..=end, target_state: Identifier { id }}
         };
         let links = vec.iter().cloned().map(link).collect();
         State {links,..Default::default()}
     }
 }
+
+
+
+// ================
+// == Identifier ==
+// ================
+
+/// A state identifier for an arbitrary finite automaton.
+#[derive(Clone,Copy,Debug,PartialEq,Eq,PartialOrd,Ord,Hash)]
+pub struct Identifier {
+    #[allow(missing_docs)]
```

Comment:
move this to struct instead. This makesthe code (Arguably) more clean - we use it this way everywhere in the codebase.